docs/how-to/craft/add-a-snap-configuration.rst

```diff
@@ -0,0 +1,222 @@
+.. _how-to-add-a-snap-configuration:
+
+Add a snap configuration
+========================
+
+This guide describes how to set up :ref:`configuration options
+<explanation-snap-configurations>` in your snap.
+
+
+Review the requirements
+-----------------------
+
+In a standard setup for configuration options, you create four scripts:
+
+- A **wrapper script** calls the app with the necessary environment variables
+- A **management script** sets values, checks their validity, and restarts the app
+- An optional **default-configure hook script** calls the management functions on first
+  install before services are started
+- A **configure hook script** calls the management functions on installation and changes
```

Comment:
@mr-cal The procedures follow this order, and I'm not a fan of it. It doesn't seem very intuitive to me as to how the whole thing is cascading.

What do you think about changing it to:

1. configure hook
2. default-configure hook
3. Wrapper script
4. Management script

Reply:
Agreed, I like your proposed order better.

Reply:
Reordered: caa4b1a